crates/router/src/core/payouts/validator.rs

```diff
@@ -84,6 +84,11 @@ pub async fn validate_create_request(
     Option<domain::Customer>,
     Option<PaymentMethod>,
 )> {
+    if req.payout_method_id.is_some() && req.confirm == Some(false) {
+        return Err(report!(errors::ApiErrorResponse::InvalidRequestData {
+            message: "Cannot set confirm to false when using stored payout method data".to_string(),
+        }));
+    }
```

Comment:
It should be checked against `Some(true)` since confirm can be `None` too

```suggestion
    if req.payout_method_id.is_some() && req.confirm != Some(true) {
        return Err(report!(errors::ApiErrorResponse::InvalidRequestData {
            message: "Confirm must be true for recurring payouts".to_string(),
        }));
    }
```